meson.build

```diff
@@ -0,0 +1,164 @@
+project('ratbagd', 'c',
+	version : '0.2',
+	license : 'MIT/Expat',
+	meson_version : '>= 0.40.0')
+
+ratbagd_version = meson.project_version().split('.')
+
+# Compiler setup
+cc = meson.get_compiler('c')
+cflags = [
+        '-pipe',
+        '-Wall',
+        '-Wextra',
+        '-Wno-inline',
+        '-Wundef',
+        '-Wformat=2',
+	'-Wformat-security',
+	'-Wformat-nonliteral',
+        '-Wlogical-op',
+        '-Wsign-compare',
+        '-Wmissing-include-dirs',
+        '-Wold-style-definition',
+        '-Wpointer-arith',
+        '-Winit-self',
+        '-Wdeclaration-after-statement',
+        '-Wfloat-equal',
+        '-Wsuggest-attribute=noreturn',
+        '-Wmissing-prototypes',
+        '-Wstrict-prototypes',
+        '-Wredundant-decls',
+        '-Wmissing-declarations',
+        '-Wmissing-noreturn',
+        '-Wendif-labels',
+        '-Wstrict-aliasing=2',
+        '-Wwrite-strings',
+        '-Wno-long-long',
+        '-Wno-overlength-strings',
+        '-Wno-unused-parameter',
+        '-Wno-missing-field-initializers',
+        '-Wno-unused-result',
+        '-Werror=overflow',
+        '-Wdate-time',
+        '-Wnested-externs',
+        '-ffast-math',
+        '-fno-common',
+        '-fdiagnostics-show-option',
+        '-fno-strict-aliasing',
+        '-fvisibility=hidden',
+        '-ffunction-sections',
+        '-fdata-sections',
+        '-fstack-protector',
+        '-fstack-protector-strong',
+        '-fPIE',
+        '--param=ssp-buffer-size=4'
+]
```

Comment:
Legacy, that originated in systemd code... The only ones I use these days are `-fno-common` and `-fvisibility=hidden`. I think everything else can be safely dropped. Especially the warnings belong into the environment of the developer, not the upstream build instructions.